q2_cutadapt/tests/test_demux.py

```diff
@@ -158,7 +161,62 @@ def test_variable_length_barcodes(self):
         self.assert_untrimmed_results(b'', obs_untrimmed_art)
 
 
-class TestDemuxUtils(TestPluginBase):
+class TestDemuxPaired(TestPluginBase):
+    package = 'q2_cutadapt.tests'
+
+    def assert_demux_results(self, exp_samples_and_barcodes, obs_demuxed_art):
+        obs_demuxed = obs_demuxed_art.view(
+            SingleLanePerSamplePairedEndFastqDirFmt)
+        obs_demuxed_seqs = obs_demuxed.sequences.iter_views(FastqGzFormat)
+        # Since we are working with fwd/rev reads, duplicate each list elem
+        exp = [x for x in exp_samples_and_barcodes.iteritems() for _ in (0, 1)]
```

Comment:
I think this might be the same:
```python
exp = list(exp_samples_and_barcode.iteritems()) * 2
```

Reply:
nevermind. nested list comprehensions kill my brain every time I see them.

Reply:
I don't love the idea of the nested list comprehensions, but, for the sake of the test this was easy enough. I can rework to something more explicit if that would be helpful.